Description: Deduplicate Menu All patch
 This commit has been rebased on the 10.4 bionic code base.
Author: Budgie Desktop Developers
Origin: Commit 3f693c254117b1a64ea65ffa22e881d0b7a03828
Last-Update: 2018-12-20
---
 src/session/meson.build | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

Index: budgie-desktop-10.4+git20171031.10.g9f71bb8/src/applets/budgie-menu/BudgieMenuWindow.vala
===================================================================
--- budgie-desktop-10.4+git20171031.10.g9f71bb8.orig/src/applets/budgie-menu/BudgieMenuWindow.vala
+++ budgie-desktop-10.4+git20171031.10.g9f71bb8/src/applets/budgie-menu/BudgieMenuWindow.vala
@@ -418,6 +418,7 @@ public class BudgieMenuWindow : Budgie.P
                 categories_scroll.set_visible(vis);
                 compact_mode = vis;
                 content.invalidate_headers();
+                content.invalidate_filter();
                 content.invalidate_sort();
                 break;
             case "menu-headers":
@@ -429,6 +430,7 @@ public class BudgieMenuWindow : Budgie.P
                     content.set_header_func(null);
                 }
                 content.invalidate_headers();
+                content.invalidate_filter();
                 content.invalidate_sort();
                 break;
             case "menu-categories-hover":
@@ -594,11 +596,11 @@ public class BudgieMenuWindow : Budgie.P

         // No more filtering, show all
         if (group == null) {
-            // Don't dupe when in "super compact" mode
-            if (super_compact && this.is_item_dupe(child)) {
-                return false;
+            if (this.headers_visible) { // If we are going to be showing headers
+                return true;
+            } else { // Not showing headers
+                return !this.is_item_dupe(child);
             }
-            return true;
         }

         // If the GMenu.TreeDirectory isn't the same as the current filter, hide it
